### PR TITLE
Temporarily remove profiler (de)activation metric logging

### PR DESCRIPTION
**Pull request by NuojCheng**
Introduce a new config `hide_profiler_step_metric` to omit metric logging for profiler start/stop and subsequent steps.

This change temporarily removes metric logging for four specific steps:
1.  The step where the profiler is activated.
2.  The step immediately following profiler activation.
3.  The step where the profiler is deactivated.
4.  The step immediately following profiler deactivation.

These steps produce outlier timings because profiler activation/deactivation overhead is currently included in the step time, which skews MFU and other benchmark metrics. This is a short-term fix to address the benchmarking inaccuracies described in b/456828037.

A long-term solution, moving the profiler to a separate subprocess to isolate its overhead, is planned.

# Tests

1. Tested with the command:
`smoke_train skip_first_n_steps_for_profiler=10 profiler_steps=3 profiler=xplane steps=20`

The log difference can be viewed here: https://diff.googleplex.com/#key=JIA9QUANVfdW

2. Tested with the command:
`smoke_train skip_first_n_steps_for_profiler=10 profiler_steps=3 profiler=xplane steps=20 hide_profiler_step_metric=true`

The log difference can be viewed here: https://diff.googleplex.com/#key=LbBZkoUsDAW5

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: src/MaxText/metric_logger.py [hunk 1]
```diff
@@ -104,43 +104,80 @@ def write_metrics(self, metrics, step, is_training=True):
...
-        )
-
-    max_logging.log(log_message)
+      self._log_eval_metrics(metrics, step)
+
+  def _log_training_metrics(self, metrics, step):
+    """Handles training-specific metric logging."""
+    # Skip logging if in profiler activation/deactivation steps
+    # TODO(b/456828037): Switch to subprocess profiling to avoid timing artifacts at boundary steps.
+    if self._is_profiler_boundary_step(step):
```
Comment (Obliviour) [inline src/MaxText/metric_logger.py]: I think its worth adding a log saying "completed step: {step}  skipping step metrics due to profilier activation/deactivation"

Comment (NuojCheng) [reply src/MaxText/metric_logger.py]: Updated! Not sure whether Ubench they use regex to grab MFU info from this or not. I am worried that outputs with different format may break their logic. Will check with them next week

**Review by bvandermoon — APPROVED** [review]

File: src/MaxText/metric_logger.py [hunk 2]
```diff
@@ -104,43 +104,93 @@ def write_metrics(self, metrics, step, is_training=True):
...
+      self._log_eval_metrics(metrics, step)
+
+  def _log_training_metrics(self, metrics, step):
+    """Handles training-specific metric logging."""
+    # Skip logging if in profiler activation/deactivation steps
+    # TODO(b/456828037): Switch to subprocess profiling to avoid timing artifacts at boundary steps.
+    scalars = metrics["scalar"]
+    loss = scalars["learning/loss"]
+    is_rampup = step < self.config.rampup_end_step
+    is_metric_hidden_step = self.config.hide_profiler_step_metric & self._is_profiler_boundary_step(step)
```
Comment (gagika) [inline src/MaxText/metric_logger.py]: Could you use `and` instead of `&` for boolean and?

e.g. either

self.config.hide_profiler_step_metric and self._is_profiler_boundary_step(step)

or 
bool(self.config.hide_profiler_step_metric) and self._is_profiler_boundary_step(step)

Comment (NuojCheng) [reply src/MaxText/metric_logger.py]: Good catch! Updated. Thank you Gagik.
